crates/sui-core/src/authority_server.rs

```diff
@@ -954,7 +984,14 @@ fn normalize(err: SuiError) -> Weight {
 macro_rules! handle_with_decoration {
     ($self:ident, $func_name:ident, $request:ident) => {{
         if $self.client_id_source.is_none() {
-            return $self.$func_name($request).await;
+            return match $self.$func_name($request).await {
+                Ok((result, _)) => {
+                    Ok(result)
+                }
+                Err(err) => {
+                    Err(err)
+                }
```

Comment:
this could just be `.map(|(result, _)| result)`

Reply:
Fixed